core/src/main/scala/java/text/SimpleDateFormat.scala

```diff
@@ -10,6 +10,67 @@ class SimpleDateFormat(private[this] val pattern: String,
 
   //def this() = this("???", DecimalFormatSymbols.getInstance())
 
+  def format(date: Date): String = {
+    val fields = Array(
+      (date.getYear + 1900).toString.reverse,
+      (date.getMonth + 1).toString.reverse,
+      date.getDate.toString.reverse,
+      date.getHours.toString.reverse,
+      date.getMinutes.toString.reverse,
+      date.getSeconds.toString.reverse,
+      (date.getTime - Date.UTC(date.getYear,
+         date.getMonth,
+         date.getDay,
+         date.getHours,
+         date.getMinutes,
+         date.getSeconds)).toString.reverse
+    )
+
+    def pop(pos: Int): Char = {
+      if (fields(pos).size > 0) {
+        val res = fields(pos)(0)
+        fields(pos) = fields(pos).drop(1)
+        res
+      } else '0'
+    }
+
+    var patternPos = pattern.length
+
+    val sb = new StringBuffer()
+    while (patternPos > 0) {
+      patternPos -= 1
+
+      pattern(patternPos) match {
+        case 'G' => ???
```

Comment:
laziness/time :-) to get this right extensively I believe it requires a lot of effort...